react/features/noise-suppression/actions.ts

```diff
@@ -0,0 +1,44 @@
+import { SET_NOISE_SUPPRESSION_STATE, TOGGLE_NOISE_SUPPRESSION } from './actionTypes';
+import { isNoiseSuppressionActive } from './functions';
+
+/**
+ * Updates the noise suppression active state.
+ *
+ * @param {boolean} active - Is noise suppression active.
+ * @returns {{
+ *      type: SET_NOISE_SUPPRESSION_STATE,
+ *      active: boolean
+ * }}
+ */
+export function setNoiseSuppressionState(active: boolean) : any {
+    return {
+        type: SET_NOISE_SUPPRESSION_STATE,
+        active
+    };
+}
+
+/**
+ * Signals that noise suppression action has been triggered.
+ *
+ * @returns {{
+ *     type: TOGGLE_NOISE_SUPPRESSION,
+ * }}
+ */
+export function toggleNoiseSuppression() : any {
+    return {
+        type: TOGGLE_NOISE_SUPPRESSION
+    };
+}
+
+/**
+ * Turns off noise suppression if it's already enabled
+ *
+ * @returns {Function}
+ */
+ export function turnOffNoiseSuppression() : Function {
+    return async (dispatch: Function, getState: Function) => {
+        if (isNoiseSuppressionActive(getState())) {
+            await dispatch(toggleNoiseSuppression());
```

Comment:
removed.

Reply:
I've added this back, it seems that even though it's not a thunk that's being returned, dispatch will return a promise which awaits for the completion of the async middleware (I think). In this case it awaits for the NS middleware to finish the setEffect(undefined) action. We need to wait for it to finish otherwise any subsequent effects set on that track might throw a 'setEffect change already in progress error'.
The case I'm trying to address here is :
1.  User has noise suppression activated.
2. User start audio screen sharing.
3. Audio screen sharing implies adding a different effect to the current local track, because with the current implementation we can't have multiple effects on a single track we need to remove the ns effect, this is done via the turnOffNoiseSuppression that will remove it in the associated middleware and update the state accordingly. Basically we need to wait for the async middleware called by the dispatch to finish, and it seems await does the trick.
An alternative would be to not allow the user to start audio sharing while NS is active, but this flow felt more natural. Even if we were to use the alternative flow, the await here seems to have an effect, do you know why? or is there a different approach we could take if this feels hacky?